src/front/app/features/challenge/challenge.reducer.js

```diff
@@ -12,8 +14,12 @@ const reducer = (state = initialState, action = {}) => {
         return upsertChallenge(action.challenge, state);
     case RESET_STATE:
         return initialState;
-    case GET_USER:
+    case USER_RECEIVED:
         return upsertCurrentUser(action.user, state);
+    case ACCEPTED_RECEIVED:
+        return state.setIn(['currentUser', 'acceptedChallenges'], fromJS(action.challenges));
+    case WATCHLIST_RECEIVED:
+        return state.setIn(['currentUser', 'watchList'], fromJS(action.challenges));
     case ADDED_TO_WATCHLIST:
         return state.set('isWatched', true);
```

Comment:
@borys-ihnatyev, we can do it the way you propose, but doing all this on backend is a completely different approach. And I'm not sure whether I've got what you mean by

> rewrite whole state on add_to_watch_list and accept_challenge

because as far as I understand `state.set()` does return completely new state